tests/expr_and_series/median_test.py

```diff
@@ -36,3 +37,32 @@ def test_median_series(
     series = nw.from_native(constructor_eager(data), eager_only=True)[col]
     result = series.median()
     assert_equal_data({col: [result]}, {col: [expected]})
+
+
+@pytest.mark.parametrize("expr", [nw.col("s").median(), nw.median("s")])
+def test_median_expr_raises_on_str(
+    constructor: Constructor,
+    expr: nw.Expr,
+    request: pytest.FixtureRequest,
+) -> None:
+    if "polars" in str(constructor):
+        request.applymarker(pytest.mark.xfail)
```

Comment:
Disclaimer: I am in a bit of a rush, I hope this doesn't get too sloppy of a comment.

You could use a similar approach as in `drop_test`, namely:

- Import polars exception: 
    ```py
    from polars.exceptions import InvalidOperationError as PlInvalidOperationError
    ```
- Match both in raise:
    ```py
    with pytest.raises((InvalidOperationError, PlInvalidOperationError), match=...):
    ```
    I think error message may differ a bit, but you can partially match it
- Regarding polars lazy, you can add a lazy+collect step to trigger the computation, hence the exception:
     ```py
     df.select(expr).lazy().collect()
     ```

I hope this helps 😇

Reply:
Hi @FBruzzesi, thanks for your help and sorry for the late reply, I was on holiday. Just to wrap up and to verify whether my understanding is correct: we might then accept raising the custom exception where feasible (or easier to manage**), else we might accept raising the original exception. Is my understanding correct?

** or, at least, it might be me not having found an easy way to handle it.

Reply:
> sorry for the late reply

No need to be sorry! It's all well and good :)

> we might then accept raising the custom exception where feasible

Yes correct, that's exactly my point, and how we already do for `DataFrame.drop` method and `ColumnNotFoundError` exception